scm/github.go

```diff
@@ -237,6 +237,24 @@ func (s *GithubSCM) UpdateRepoAccess(ctx context.Context, repo *Repository, user
 	return err
 }
 
+// RepositoryIsEmpty implements the SCM interface
+func (s *GithubSCM) RepositoryIsEmpty(ctx context.Context, opt *RepositoryOptions) bool {
+
+	repo, err := s.GetRepository(ctx, opt)
+	if err != nil {
+		return false
+	}
+
+	// test to check how repo commits look like
+	_, _, err = s.client.Repositories.ListCommits(ctx, repo.Owner, repo.Path, nil)
+	if err != nil {
+		if strings.Contains(err.Error(), "Git Repository is empty") {
```

Comment:
This is a bit fragile to check for such a string, but I understand why. I googled a bit, and came across this [stackoverflow issue](https://stackoverflow.com/questions/19889899/count-number-of-empty-repositories-on-github). Specifically, this comment:

_That would be an interesting statistic. We don't have a simple way to do that right now. However, you might be able to use the GitHub API to get close. You could look through public repositories and compare "pushed_at" and "created_at" dates to see if there has been any activity. Additionally, you could find repositories with a "size" of 0. There's more information on how to find this information, and much more, right here:_

http://developer.github.com/v3/repos/

Would this give us the same in a safer way?

Reply:
Unfortunately, none of these options work.
A repo with a few text files pushed to it still has size 0.
For an empty repo `pushed_at` can somehow differ from `created_at` (Example: Repository  grp1  was created at:  2019-11-14 13:54:43 +0000 UTC  and last pushed to at: 2019-11-14 13:54:44 +0000 UTC).